### PR TITLE
File logging

### PR DESCRIPTION
**Pull request by miiu96**
- Moved the part where the logs files are handled from `elrond-go` repository.

### CODE REVIEW

File: file/errorLogging.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package file
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+)
+
+// LogErrAsWarnExceptAsDebugIfClosingError logs an error
```
Comment (iulianpascalau) [inline file/errorLogging.go:8]: I would keep this in elrond-go only. Is too specific

File: file/errors.go [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	errContextClosing   = errors.New("context closing")
+	errInvalidParameter = errors.New("invalid parameter provided")
+)
+
+func isClosingError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	return strings.Contains(err.Error(), errDBIsClosed.Error()) ||
```
Comment (iulianpascalau) [inline file/errors.go:19]: too specific, let's put back in elrond-go

File: file/fileLogging.go [hunk 3]
```diff
@@ -0,0 +1,233 @@
+package file
```
Comment (iulianpascalau) [inline file/fileLogging.go:1]: diff checked, looks good :+1:

File: file/fileLogging_test.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
+package file
```
Comment (iulianpascalau) [inline file/fileLogging_test.go:1]: diff checked, looks good :+1:

File: file/interface.go [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package file
```
Comment (iulianpascalau) [inline file/interface.go:1]: Do we use this interface in this package? Might be moved back in elrond-go

File: file/lifeSpanner.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
+package file
```
Comment (iulianpascalau) [inline file/lifeSpanner.go:1]: diff checked, looks good :+1:

File: file/lifeSpanner_test.go [hunk 7]
```diff
@@ -0,0 +1,156 @@
+package file
```
Comment (iulianpascalau) [inline file/lifeSpanner_test.go:1]: diff checked, looks good :+1:

File: go.mod [hunk 8]
```diff
@@ -1,10 +1,21 @@
 module github.com/ElrondNetwork/elrond-go-logger
 
-go 1.13
+go 1.17
```
Comment (iulianpascalau) [inline go.mod:3]: :+1:

File: file/errorLogging.go (hunk 1 above)
Comment (miiu96) [reply file/errorLogging.go]: done

File: file/errors.go (hunk 2 above)
Comment (miiu96) [reply file/errors.go]: done

File: file/interface.go (hunk 5 above)
Comment (miiu96) [reply file/interface.go:1]: no. removed.

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**